### PR TITLE
Make tubular an installable module.

### PR DESCRIPTION
**Pull request by doctoryes**
Also, add some GoCD-specific code, including a preliminary XML config file.
All this stuff is still just for the pipeline evaluations.

@feanil @macdiesel @benpatterson @clintonb

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+setup(
+    name='tubular',
+    version='0.1',
+    description='Continuous Delivery scripts for pipeline evaluation',
+    packages=['tubular', 'tubular.gocd', 'tubular.scripts', 'tubular.scripts.github', 'tubular.scripts.hipchat'],
+    install_requires=[
+        'click',
+        'requests',
```
Comment (feanil) [inline setup.py]: What about pygithub?  How do we ensure the correct fork gets installed?

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    ],
+)
```
Comment (clintonb) [inline setup.py]: Add a newline.

File: setup.py (hunk 1 above)
Comment (clintonb) [inline setup.py]: Requirements should be versioned. Ranges are acceptable.

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+from setuptools import setup
+
+
+setup(
+    name='tubular',
+    version='0.1',
```
Comment (clintonb) [inline setup.py]: Semantic versioning: 0.0.1

**Comment by clintonb** [comment]
Unless there are test-related requirements (e.g. `pylint`) I would remove `requirements.pip`. Too often I see requirements files become out-of-date with `setup.py`. This can be completely avoided by installing the package as editable: `pip install -e .`

When test-related requirements are needed, add them to a requirements file along with, `-e .`.

File: setup.py (hunk 1 above)
Comment (doctoryes) [inline setup.py]: @clintonb Done.

File: setup.py (hunk 2 above)
Comment (doctoryes) [inline setup.py]: Done.

File: setup.py (hunk 3 above)
Comment (doctoryes) [inline setup.py]: Done.

**Comment by doctoryes** [comment]
Unfortunately, it seems that `setup.py` cannot properly install a GitHub fork. (@nedbat checked in the Python IRC channel - and I tried various things.) Do you guys know a way to do this? Hosting our own PyPI repo would work - but seems heavy-handed.

The scheme now will work with `pip install -r requirements.pip` - all requirements and tubular will install properly.

**Comment by clintonb** [comment]
@doctoryes @nedbat see http://mike.zwobble.org/2013/05/adding-git-or-hg-or-svn-dependencies-in-setup-py/.

**Comment by doctoryes** [comment]
Unfortunately, the method in the article above doesn't work - I tried it. First, `dependency_links` isn't supported unless an `pip` option is used - `--process-dependency-links`. Also, since PyGithub has an existing package in PyPI, the setup.py process looks at those PyPI versions instead of following the GitHub link provided.

**Comment by clintonb** [comment]
What's special about this fork? What are the prospects of these changes being merged into the main repo?

**Comment by doctoryes** [comment]
The fork just adds a check for combined status that's missing from the current module. I've submitted an upstream PR here:
https://github.com/PyGithub/PyGithub/pull/382

**Comment by feanil** [comment]
We should make a seperate setup.py PR and PR for scripts you are adding. Also, it looks like you're using requirements.pip here and I merged in requirements.txt already.

**Comment by doctoryes** [comment]
@feanil @macdiesel I've rebased and removed requirements.pip.

**Comment by feanil** [comment]
:+1:

**Pull request merged**